server/liveblog/syndication/utils.py

```diff
@@ -318,14 +319,25 @@ def create_syndicated_blog_post(producer_post, items, in_syndication):
                 continue
 
         item["blog"] = in_syndication["blog_id"]
-        post_items.append(item)
 
-    items_service = get_resource_service("blog_items")
-    item_ids = items_service.post(post_items)
+        if item["item_type"] == "poll":
+            post_polls.append(item)
+        else:
+            post_items.append(item)
+
     item_refs = []
 
-    for i, item_id in enumerate(item_ids):
-        item_refs.append({"residRef": str(item_id)})
+    if post_items:
+        item_service = get_resource_service("blog_items")
+        item_ids = item_service.post(post_items)
+        item_refs.extend({"residRef": str(item_id)} for item_id in item_ids)
+
+    if post_polls:
+        poll_service = get_resource_service("blog_polls")
+        poll_ids = poll_service.post(post_polls)
+        item_refs.extend(
+            {"residRef": str(poll_id), "location": "polls"} for poll_id in poll_ids
+        )
```

Comment:
This looks very redundant to me. Is there a way we can make this dynamic somehow? We will have more complex item types like polls in the future.